connection/cluster/ClusterClient.java

```diff
@@ -322,14 +336,19 @@ private void waitForPrimaryReplicaSelection() {
             }
         }
 
+        private boolean isNonExistentRaftClusterInfo(ClusterDatabase clusterDatabase) {
+            return clusterDatabase.primaryReplica().isEmpty() &&
+                    clusterDatabase.replicas().stream().allMatch(replica -> replica.term() == -1);
+        }
```

Comment:
When a replicator does not exist on a node, it returns an entry with no primary and all terms set to -1.

Reply:
Do we need the first check here?

Reply:
May not be, but it's a tighter condition.